website/blog/2024/06-24-proxy-management-in-crawlee/index.md

```diff
@@ -0,0 +1,104 @@
+
+:::note
+
+If you like reading this blog, we would be really happy if you gave [Crawlee a star on GitHub!](https://github.com/apify/crawlee/)
+
+:::
+
+## What are tiered proxies?
+
+Tiered proxies are a method of organizing and using different types of proxies based on their quality, speed, reliability, and cost. Tiered proxies allow you to rotate between a mix of proxy types to optimize your scraping activities.
+
+**Define proxy tiers**: You categorize your proxies into different tiers based on their quality. For example:
+
+- **High-tier proxies**: Fast, reliable, and expensive. Best for critical tasks where you need high performance.
+- **Mid-tier proxies**: Moderate speed and reliability. A good balance between cost and performance.
+- **Low-tier proxies**: Slow and less reliable but cheap. Useful for less critical tasks or high-volume scraping.
+
+## Features:
+
+- **Tracking errors**: The system monitors errors (e.g. failed requests, retries) for each domain.
+- **Adjusting tiers**: Higher-tier proxies are used if a domain shows more errors. Conversely, if a domain performs well with a high-tier proxy, the system will occasionally test lower-tier proxies. If successful, it continues using the lower tier, optimizing costs.
+- **Forgetting old errors**: Old errors are given less weight over time, allowing the system to adjust tiers dynamically as proxies' performance changes.
+
+### Working
+
+The `tieredProxyUrls` option in Crawlee's `ProxyConfigurationOptions` allows you to define a list of proxy URLs organized into tiers. Each tier represents a different level of quality, speed, and reliability.
+
+### Structure
+
+**Tiered Array**: Each element in the array is an object containing proxy URLs for that tier.
+
+### Usage
+
+**Fallback Mechanism**: Crawlee starts with the first tier of proxies. If proxies in the current tier fail, it will switch to the next tier.
+
+### Implementation:
+
+```js
+import { CheerioCrawler, ProxyConfiguration } from 'crawlee';
+
+const proxyConfiguration = new ProxyConfiguration({
+    tieredProxyUrls: [
+        ['http://tier1-proxy1.example.com', 'http://tier1-proxy2.example.com'],
+        ['http://tier2-proxy1.example.com', 'http://tier2-proxy2.example.com'],
+        ['http://tier2-proxy1.example.com', 'http://tier3-proxy2.example.com'],
+    ],
+});
+
+const crawler = new CheerioCrawler({
+    proxyConfiguration,
+    requestHandler: async ({ request, response }) => {
+        // Handle the request
+    },
+});
+
+await crawler.addRequests([
+    { url: 'https://example.com/critical' },
+    { url: 'https://example.com/important' },
+    { url: 'https://example.com/regular' },
+]);
+
+await crawler.run();
+```
+
+## How tiered proxies use Session Pool under the hood
```

Comment:
While this part is not wrong, the implementation of all of this in Crawlee is a mess. I don't think people profit much from knowing how it's implemented.

Reply:
the thinking was as I said earlier is to explain we do rotate/create sessions and use session pool as it is one of the feature that makes us different from scrapy and we under promote it.

and this is making the blog a little more technical than just talking about the product/crawlee update.